### DISCUSSION TITLE
Migrate from ionic-storage. Asking for some advise

### DISCUSSION THREAD

**Discussion by juanmaldonadodev** [capacitor-community/sqlite#442] [Q&A]
Currently we are using in a Ionic/Angular/Capacitor project.

ionic-storage with localforage-cordovasqlitedriver and cordova-sqlite-storage 
Also we create a second database handle without ionic-storage. We created this with an objective but currently this information could be handle as ionic-storage.

We have detected a problem with localforage-cordovasqlitedriver and check that library is not maintained or at least from last 4 years.

So what we have in mind reviewing capacitor options.

At the end we have two sqlite databases.

1. ionic-database created by ionic-storage
2. custom-database created by us handle by just using cordova-sqlite-storage 

What we would like to do will be use one of this capacitor plugins
capacitor-data-storage-sqlite or capacitor-community/sqlite

We want to migrate ionic-database and custom-database data to a new capacitor database. Data could be handle if we storage them by key, value as capacitor-data-storage-sqlite utilities can work with them.

So here some questions:

Can we use just capacitor-data-storage-sqlite and migrate information of old two databases to a the one created by this plugin?

Should we use capacitor-community/sqlite and handle migration with this one? https://github.com/capacitor-community/sqlite/blob/master/docs/MigratingCordovaDatabases.md 

Can we use capacitor-community/sqlite to handle migration but use capacitor-data-storage-sqlite to access, get and set values to migrated data?

I think it will be really helpful for others users using localforage-cordovasqlitedriver that have moved to capacitor.

Thanks

**Comment by ghenry22** [comment]
@juanico18 I've been working on the same issue trying to migrate to a capacitor solution so I can remove the old cordova-sqlite plugin.

I have tried following the guide that you linked but have had no luck.  The database list and migration commands all execute fine and I can see the new migrated DB in the list post migration but when I try to open that migrated DB using this plugin it seems to be empty.  Any query just returns an empty result for me.

If you have any luck please do post an update!

**Reply by jepiqueau** [reply]
@juanico18 are you ionic-database created by ionic-storage and your custom database created by using cordova-sqlite-storage are key,value pair data. if yes you should definitively use capacitor-data-storage-sqlite. Unfortunately i did not implement for this plugin the database list and migration commands. So there is nothing wrong to use @capacitor-community/sqlite to do the migration of the database and after use capacitor-data-storage-sqlite when done. 
@ghenry22 surprise to hear that the copied database are empty. can you gave me the location of the ionic-storage and cordova-sqlite-storage databases and on which platforms it happens so i can check

**Reply by juanmaldonadodev** [reply]
Hello @jepiqueau 
Yes I use a key, value pair data. But values can be json.
It would be ok anyway?
